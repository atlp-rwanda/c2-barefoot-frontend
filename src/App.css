.toolbar{
  display: flex;
  justify-content: space-between;
}

.menuButton{
  margin-right: 20px;
}