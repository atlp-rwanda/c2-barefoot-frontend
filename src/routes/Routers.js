import ResetPasswordEmailForm from '../components/resetPassword/ResetPasswordEmailForm';
import NewPassword from '../components/resetPassword/NewPassword'

import React from 'react';
import Login from '../components/views/Login'
import SignUp from '../components/signup';
import signup from '../components/views/Signup'
import verifyAccount from '../components/signup/verifyAccount'
import Landing from '../components/views/LandingPage'
import PageNotFound from '../components/views/PageNotFound'
import { Switch, Redirect } from 'react-router-dom';
import RouteWithLayout  from '../components/RouteWithLayout';
import TravelAdminDefault  from '../components/travelAdminLayoutRoute/Default';
import { createLocation } from '../components/views/travelAdmin/createLocations';
import { createAccomodations } from '../components/views/travelAdmin/createAccommodations';
import TravelAdmin from '../components/views/travelAdmin';
import TravelAdminRoute from '../components/travelAdminLayoutRoute';
import ListLocations from '../components/views/travelAdmin/viewLocations';
import ListAccommodation from '../components/views/travelAdmin/viewAccommodations';
// import { redirectUser, toBeRedirected } from '../../../services/userInfo';
import Profile from '../components/views/Profile';
import {  DefaultLayout, AuthorizedUserLayout } from '../components/layouts';
import Logout from '../components/views/Logout';

const Routes = () => {
    return (
      <Switch>
        <Redirect
          exact
          from="/"
          to="/welcome"
        />
        <RouteWithLayout
          component={Landing}
          exact
          layout={DefaultLayout}
          path="/welcome"
        />
        <RouteWithLayout
          component={Login}
          exact
          layout={DefaultLayout}
          path="/login"
        />
        <RouteWithLayout
          component={Profile}
          exact
          layout={AuthorizedUserLayout}
          path="/profile"
        />
         <RouteWithLayout
          component={signup}
          exact
          layout={DefaultLayout}
          path="/signup"
        />
        <RouteWithLayout
          component={ResetPasswordEmailForm}
          exact
          layout={DefaultLayout}
          path="/forgetpassword"
        />
        <RouteWithLayout
          component={NewPassword}
          exact
          layout={DefaultLayout}
          path="/user/reset-password"
        />
        <RouteWithLayout
          component={Logout}
          exact
          layout={AuthorizedUserLayout}
          path="/logout"
        />
        <RouteWithLayout
          component={PageNotFound}
          exact
          layout={DefaultLayout}
          path="/PageNotFound"
        />
        <RouteWithLayout 
            path="/signup" 
            component={signup}
            layout={DefaultLayout}
        />
        <RouteWithLayout 
            path="/user/verification/:token" 
            component={ verifyAccount }
            layout={DefaultLayout}
        />
        <RouteWithLayout 
            path="/user/verification/" 
            component={ verifyAccount }
            layout={DefaultLayout}
        />
        <RouteWithLayout 
            path="/admin/travel" 
            exact
            component={ TravelAdmin }
            layout={TravelAdminDefault}
        />
        <RouteWithLayout 
            path="/admin/travel/location/create" 
            component={ createLocation }
            layout={TravelAdminDefault}
        />
        <RouteWithLayout 
            path="/admin/travel/location/view" 
            component={ ListLocations }
            layout={TravelAdminDefault}
        />
        <RouteWithLayout 
            path="/admin/travel/accommodations/create" 
            component={ createAccomodations }
            layout={TravelAdminDefault}
        />
        <RouteWithLayout 
            path="/admin/travel/accommodations/view" 
            component={ ListAccommodation }
            layout={TravelAdminDefault}
        />
        
        <Redirect to="/PageNotFound" />
      </Switch>
    );
  };
  
  export default Routes;
