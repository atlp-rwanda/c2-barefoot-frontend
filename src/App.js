import React, { Component } from "react";
import NavBar from './components/NavBar'
import Routers from './routes/Routers'
import {BrowserRouter as Router} from 'react-router-dom'

class App extends Component{
     render() {
        return (
            <Router>
                <div className="App">
                    <NavBar/>
                    <Routers/>
                </div>
            </Router>
        )
    }
}
export default App;