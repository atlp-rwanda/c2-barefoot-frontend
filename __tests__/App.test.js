import React from 'react';
import App from '../src/App';
import Navbar from '../src/components/NavBar';
import Routers from '../src/routes/Routers';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

describe('<App />', () =>{
    let wrapper = shallow(<App />);
    it('renders without crashing', () => {
        wrapper = shallow(<App />);
    });
    it('should render one <Navbar /> and <Routers /> component', () => {
        expect(wrapper.find(Navbar)).toHaveLength(1);
        expect(wrapper.find(Routers)).toHaveLength(1);
    });

    it('should render with a class of container', () => {
        expect(wrapper.find('.container')).toHaveLength(1);
    });

    it('matches snapshot', () => {
        let tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    
})