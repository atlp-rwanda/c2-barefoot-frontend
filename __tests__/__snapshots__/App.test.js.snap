// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="container"
>
  <div>
    <div
      className="navbar"
    >
      <div
        className="logo"
      >
        <h2>
          Barefoot Normad
        </h2>
      </div>
      <div
        className="navlinks"
      >
        <ul>
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
             
            <a
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
          <li>
            <a
              href="/signup"
              onClick={[Function]}
            >
              Signup
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div
    className="home"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Greating
      </span>
    </button>
    <br />
    Hi, There.
    <p>
      Guys enjoy your coding.
    </p>
  </div>
</div>
`;
